plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -239,6 +239,12 @@ public Optional<ParameterizedExpression> convertPredicate(ConnectorSession sessi
         return delegate.convertPredicate(session, expression, assignments);
     }
 
+    @Override
+    public Optional<ParameterizedExpression> convertPredicateForJoin(ConnectorSession session, ConnectorExpression expression, Map<String, ColumnHandle> assignments)
```

Comment:
Is it something we could restrict the scope inside `applyJoin` itself ?

Reply:
I think it would be a bit tricky -

Reply:
I didn't find prettier solution.

Reply:
Having different version or ConnectorRewrite for each predicate like filter or join or aggregation might make the framework brittle. Can we implement something like `isSupportedJoinCondition` for the newer join framework ?  Plus doing this change just for one connector i.e Clickhouse doesn't support inequality joins as of now might also be tranisient as Clickhouse might resolve them sometime in the future. This code might loose its importance once it has been implemented.